src/pytorch_lightning/loggers/mlflow.py

```diff
@@ -204,7 +214,7 @@ def experiment_id(self) -> str:
             The experiment id.
         """
         _ = self.experiment
-        return self._experiment_id
+        return self._experiment_id  # type: ignore[return-value]
```

Comment:
Same comment applies here

Reply:
pyproject.toml can be updated to include [`strict_optional = "False"`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict_optional) under [tool.mypy]; which solves this issue where mypy seems to be checking both options ... however, updating the config causes additional errors in other files. 

```sh
src/pytorch_lightning/trainer/connectors/signal_connector.py:153: error: Unused "type: ignore" comment
src/pytorch_lightning/trainer/connectors/accelerator_connector.py:589: error: Unused "type: ignore" comment
src/pytorch_lightning/trainer/connectors/accelerator_connector.py:667: error: Unused "type: ignore" comment
src/pytorch_lightning/trainer/connectors/accelerator_connector.py:669: error: Unused "type: ignore" comment
src/pytorch_lightning/trainer/connectors/accelerator_connector.py:682: error: Unused "type: ignore" comment
src/pytorch_lightning/trainer/connectors/accelerator_connector.py:699: error: Unused "type: ignore" comment
src/pytorch_lightning/trainer/connectors/accelerator_connector.py:755: error: Unused "type: ignore" comment
src/pytorch_lightning/strategies/single_hpu.py:65: error: Unused "type: ignore" comment
src/pytorch_lightning/strategies/fully_sharded_native.py:193: error: Unused "type: ignore" comment
src/pytorch_lightning/strategies/bagua.py:148: error: Unused "type: ignore" comment
src/pytorch_lightning/strategies/bagua.py:149: error: Unused "type: ignore" comment
src/pytorch_lightning/strategies/launchers/spawn.py:118: error: Unused "type: ignore" comment
```

Reply:
I don't understand why that would help. We want to be strict here.

Reply:
please check latest commit